vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java

```diff
@@ -41,12 +42,31 @@ public ArrayTuple(Collection<?> c) {
 
   @Override
   public <T> T get(Class<T> type, int pos) {
-    throw new UnsupportedOperationException();
+    Object value = this.get(pos);
+    if (value.getClass() == type) {
```

Comment:
check nullity and add a test for it

Reply:
added in https://github.com/eclipse-vertx/vertx-sql-client/pull/367/commits/4beca16bc56cc291c3e8f0b73d3c9517dc3c5055.